tests/testDetectAndMeasure.py

```diff
@@ -0,0 +1,134 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""Test DetectAndMeasureTask
+"""
+import os
+import unittest
+
+import lsst.utils.tests
+from lsst.utils import getPackageDir
+from lsst.afw.geom import Box2I, Point2I, Extent2I
+from lsst.afw.table import SourceTable
+from lsst.daf.persistence import Butler
+from lsst.meas.algorithms.testUtils import plantSources
+from lsst.pipe.tasks.detectAndMeasure import DetectAndMeasureTask
+
+try:
+    obsTestDir = getPackageDir("obs_test")
+    InputDir = os.path.join(obsTestDir, "data", "input")
+except Exception:
+    InputDir = None
+
+class ProcessCcdTestCase(lsst.utils.tests.TestCase):
+    def testBasics(self):
+        """Test detection and measurement on simple synthesized data
+        """
+        bbox = Box2I(Point2I(256, 100), Extent2I(128, 127))
+        minCounts = 5000
+        maxCounts = 50000
+        starSigma = 1.5
+        numX = 5
+        numY = 5
+        coordList = self.makeCoordList(
+            bbox = bbox,
+            numX = numX,
+            numY = numY,
+            minCounts = minCounts,
+            maxCounts = maxCounts,
+            sigma = starSigma,
+        )
+        kwid = 11 # kernel width
+        sky = 2000
+        # create an exposure without a Wcs; add the Wcs later
+        exposure = plantSources(bbox=bbox, kwid=kwid, sky=sky, coordList=coordList, addPoissonNoise=True)
+
+        schema = SourceTable.makeMinimalSchema()
+
+        config = DetectAndMeasureTask.ConfigClass()
+        task = DetectAndMeasureTask(config=config, schema=schema)
+
+        butler = Butler(root=InputDir)
+        dataRef = butler.dataRef("calexp", dataId=dict(visit=1))
+        wcs = dataRef.get("raw").getWcs()
+        exposure.setWcs(wcs)
+        exposureIdInfo = dataRef.get("expIdInfo")
+        taskRes = task.run(exposure=exposure, exposureIdInfo=exposureIdInfo)
+        self.assertEqual(len(taskRes.sourceCat), numX * numY)
+        schema = taskRes.sourceCat.schema
+        centroidFlagKey = schema.find("slot_Centroid_flag").getKey()
+        parentKey = schema.find("parent").getKey()
+        psfFluxFlagKey = schema.find("slot_PsfFlux_flag").getKey()
+        psfFluxKey = schema.find("slot_PsfFlux_flux").getKey()
+        for src in taskRes.sourceCat:
+            self.assertFalse(src.get(centroidFlagKey))  # centroid found
+            self.assertEqual(src.get(parentKey), 0)     # not debelended
```

Comment:
Is this a statement about the nature of the image, because deblending should have been performed (it defaults to `True`, and I don't see it disabled)?